cpp/velox/compute/VeloxParquetDatasource.cc

```diff
@@ -49,29 +49,43 @@ void VeloxParquetDatasource::init(const std::unordered_map<std::string, std::str
   pool_ = veloxPool->addLeafChild("velox_parquet_write");
 
   // Construct the file path and writer
-  std::string localPath = "";
+  auto pos = filePath_.find("_temporary", 0);
+  std::string destinationPathWithSchame = filePath_.substr(0, pos - 1);
+
   if (strncmp(filePath_.c_str(), "file:", 5) == 0) {
-    localPath = filePath_.substr(5);
-  } else {
-    throw std::runtime_error("The path is not local file path when Write data with parquet format in velox backend!");
-  }
+    finalPath_ = (destinationPathWithSchame + "/" + fileName_).substr(5);
+    std::string command = "touch " + finalPath_;
+    auto ret = system(command.c_str());
+    if (ret != 0) {
+      throw std::runtime_error(
+          "The local file path was not created successfully when writing parqetut data in velox backend!");
+    }
 
-  auto pos = localPath.find("_temporary", 0);
-  std::string dirPath = localPath.substr(0, pos - 1);
+    sink_ = std::make_unique<velox::dwio::common::LocalFileSink>(finalPath_);
+  } else if (strncmp(filePath_.c_str(), "hdfs:", 5) == 0) {
+    finalPath_ = destinationPathWithSchame + "/" + fileName_; // with hdfs://vsr246:9000
+    auto destinationPathPos = destinationPathWithSchame.substr(7).find("/", 0);
+    std::string destinationPath = finalPath_.substr(destinationPathPos + 7);
+    std::string command = "hadoop fs -touch " + destinationPath;
```

Comment:
this seems a workaround

Reply:
Can we let it be done by client API? I think we should not assume `hadoop` is available in user's env.